src/main/java/com/ibm/eventstreams/kafkaconnect/plugins/xml/engines/StructToXmlBytes.java

```diff
@@ -233,19 +235,24 @@ private void processStruct(Document doc, Element parentElement, Struct source) {
 
                 case BYTES:
                     final Element bytesElement = doc.createElement(field.name());
-                    if (source.getBytes(field.name()) != null) {
-                        final byte[] bytes = source.getBytes(field.name());
-
-                        String strRepresentation;
-                        if (bytes.length == 1 && "xs:byte".equals(fieldSchema.doc())) {
-                            strRepresentation = Byte.toString(bytes[0]);
-                        }
-                        else {
-                            strRepresentation = new String(Base64.getEncoder().encode(bytes));
-                        }
-
-                        bytesElement.appendChild(doc.createTextNode(strRepresentation));
+                    final byte[] bytes ;
+                    String strRepresentation;
+
+                    if (Decimal.LOGICAL_NAME.equals(fieldSchema.name())){
+                       bytes = Decimal.fromLogical(fieldSchema, new BigDecimal(String.valueOf(source.get(field.name()))));
```

Comment:
These two lines are overly complex and I think a little bit circular. 

The aim of this code is to start with a `BigDecimal` object (`source.get(field.name())`) and turn it into a String to add to the XML element. 

The code here is doing that in X steps

**Step 1** - wrap it in `String.valueOf`
```java
String.valueOf(theBigDecimal)
```

**Step 2** - put the result in a constructor to turn it back into a BigDecimal
```java
new BigDecimal(String.valueOf(theBigDecimal))
```

At this point, I think the code has done a round-trip and ended up with the value it started with. And it continues.

**Step 3** - use a Connect helper method to turn a BigDecimal into a byte array
```java
byte[] bytes = Decimal.fromLogical(schema, 
                                   new BigDecimal(String.valueOf(theBigDecimal)));
``` 

**Step 4** - use the inverse Connect helper method to turn a byte array back into a BigDecimal
```java
Decimal.toLogical(schema, bytes)
```
 
At this point, I think the code has now done another round-trip and ended up with the BigDecimal value it started with again. 

**Step 5** - use toString() to get a string version
```java
Decimal.toLogical(schema, bytes).toString()
```


**In summary...** I think this code would produce an identical result if it was simplified to skip steps 1-4 and just do step 5: 

```java
strRepresentation = source.get(field.name()).toString();
```

Reply:
Thank you for the suggestion , updated accordingly